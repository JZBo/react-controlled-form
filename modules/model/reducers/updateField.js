/* @flow */
import composeFieldData from '../../utils/composeFieldData'

import type { Action } from '../../../types/Action'

export default function updateField(
  state: Object,
  {
    payload: {
      formId,
      fieldId,
      value,
      isEnabled,
      isRequired,
      isTouched,
      isValid
    }
  }: Action
): Object {
  return {
    ...state,
    [formId]: {
      ...state[formId],
      [fieldId]: composeFieldData(state[formId][fieldId], {
        isEnabled,
        isRequired,
        isValid,
        isTouched: isTouched || true,
        value
      })
    }
  }
}
