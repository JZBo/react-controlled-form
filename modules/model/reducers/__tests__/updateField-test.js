import updateField from '../updateField'

const payload = p => ({ payload: p })
const initialFormState = {
  data: {
    bar: {
      isValid: true,
      isRequired: false,
      isEnabled: true,
      value: '',
      isTouched: false,
    },
  },
  state: {},
}

describe('Updating form fields', () => {
  it('should update values and set isTouched', () => {
    expect(
      updateField(
        { foo: initialFormState },
        payload({ formId: 'foo', fieldId: 'bar', value: 'test' })
      )
    ).toMatchSnapshot()
  })
})
