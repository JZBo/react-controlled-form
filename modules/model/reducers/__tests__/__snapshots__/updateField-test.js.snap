// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Updating form fields should update values and set isTouched 1`] = `
Object {
  "foo": Object {
    "data": Object {
      "bar": Object {
        "isEnabled": true,
        "isRequired": false,
        "isTouched": true,
        "isValid": true,
        "value": "test",
      },
    },
    "state": Object {},
  },
}
`;
